fastapi/app/main.py

```diff
@@ -0,0 +1,231 @@
+    except Exception as e:
+        body = ErrorModel(error=str(e)).model_dump(exclude_none=True)
+        return ResponseModel(statusCode=500, body=body)
+
+@app.get("/getTargetFrameInfo")
+async def getTargetFrameInfo(
+    targetId: int,
+    mission: str,
+    searchKernels: bool = False):
+    try:
+        result = pyspiceql.getTargetFrameInfo(targetId, mission, searchKernels)
+        body = ResultModel(result=result).model_dump(by_alias=True, exclude_none=True)
+        return ResponseModel(statusCode=200, body=body)
+    except Exception as e:
+        body = ErrorModel(error=str(e)).model_dump(exclude_none=True)
+        return ResponseModel(statusCode=500, body=body)
+
+@app.get("/findMissionKeywords")
+async def findMissionKeywords(
+    key: str,
+    mission: str,
+    searchKernels: bool = False):
+    try:
+        result = pyspiceql.findMissionKeywords(key, mission, searchKernels)
+        body = ResultModel(result=result).model_dump(by_alias=True, exclude_none=True)
+        return ResponseModel(statusCode=200, body=body)
+    except Exception as e:
+        body = ErrorModel(error=str(e)).model_dump(exclude_none=True)
+        return ResponseModel(statusCode=500, body=body)
+
+@app.get("/findTargetKeywords")
+async def findTargetKeywords(
+    key: str,
+    mission: str,
+    searchKernels: bool = False):
+    try:
+        result = pyspiceql.findTargetKeywords(key, mission, searchKernels)
+        body = ResultModel(result=result).model_dump(by_alias=True, exclude_none=True)
+        return ResponseModel(statusCode=200, body=body)
+    except Exception as e:
+        body = ErrorModel(error=str(e)).model_dump(exclude_none=True)
+        return ResponseModel(statusCode=500, body=body)
+
+@app.get("/frameTrace")
+async def frameTrace(
+    et: float,
+    initialFrame: int,
+    mission: str,
+    ckQuality: str = "",
+    searchKernels: bool = False):
+    try:
+        result = pyspiceql.frameTrace(et, initialFrame, mission, ckQuality, searchKernels)
+        body = ResultModel(result=result).model_dump(by_alias=True, exclude_none=True)
+        return ResponseModel(statusCode=200, body=body)
+    except Exception as e:
+        body = ErrorModel(error=str(e)).model_dump(exclude_none=True)
+        return ResponseModel(statusCode=500, body=body)
+    
+@app.get("/extractExactCkTimes")
+async def extractExactCkTimes(
+    observStart: float,
+    observEnd: float,
+    targetFrame: int,
+    mission: str,
+    ckQuality: str = "",
+    searchKernels: bool = False):
+    try:
+        result = pyspiceql.extractExactCkTimes(observStart, observEnd, targetFrame, mission, ckQuality, searchKernels)
+        body = ResultModel(result=result).model_dump(by_alias=True, exclude_none=True)
+        return ResponseModel(statusCode=200, body=body)
+    except Exception as e:
+        body = ErrorModel(error=str(e)).model_dump(exclude_none=True)
+        return ResponseModel(statusCode=500, body=body)
```

Comment:
Curious - Is this amount of try/except boilerplate with consistent excepts and slightly altered try blocks standard for these types of APIs? I haven't seen enough to know how much boilerplate to expect.

Reply:
Mm just from experience, whether creating REST APIs in Java or Python, it's been best practice to wrap almost every endpoint in a try except/catch block. I think ideally there would be multiple except blocks to catch various HTTP errors and the service can return a more customized response specific to that error. However, the goal of this FastAPI app has been to replicate the same response payload as the current SpiceQL web version which currently catches any error and returns the same payload of status code 500 and the error message. I believe it is so we can test the deployed instance of the app as is without changing anything on the client's end (like ALE).

Reply:
100% agree on the try/except/finally paradigm in these. The more verbose the failure, the better for everyone (caller, log parser while debugging, etc.). 

And yeah, it was more a does something exist to reduce the boilerplate or is this it. Again, no need to do a bunch of research, I'm just reviewing as a learner.

Reply:
Should be possible to mimic how the old API did it? Basically use a decorator. 

```python
def create_lambda_call(func):
    def lambda_wrapper(event, context):
      try: 
        print("Lambda Event: ", event)
        print("Lambda Context: ", context)
        ret = {"return": func(**event)}
        print(ret)
        return {
            "statusCode" : 200,
            "body" : ret
        }
        
      except Exception as e: 
        return {
            "statusCode" : 500,
            "body" : {
                "error" : str(e)
            }
        }
    return lambda_wrapper

lambda_strSclkToEt = create_lambda_call(pyspiceql.strSclkToEt)
lambda_doubleSclkToEt = create_lambda_call(pyspiceql.doubleSclkToEt)
lambda_utcToEt = create_lambda_call(pyspiceql.utcToEt)
lambda_translateNameToCode = create_lambda_call(pyspiceql.translateNameToCode)
lambda_translateCodeToName = create_lambda_call(pyspiceql.translateCodeToName)

```

Reply:
Is there a preference to having the API mimic the old lambda format? I would have to look into the decorator thing, would it align with FastAPI guidelines (like how each endpoint has an `@app.get()` decorator)? I think I was going with the new format in possibilities of future additions or modifications, giving this format more flexibility in terms of custom payloads, messages/errors, request handling per endpoint.

Reply:
I personally think it's whatever, mostly answering Jay's question